### PR TITLE
Hell logic changes and Generalize tricks

### PR DESCRIPTION
**Pull request by dragonc0**
add some hell logic changes in separate commit.

Rewrite logic into trick variables for higher level logics, to generalize code and improve readability. 
Include text_clip variable to allow logic to check if nag messages are enabled or not for advanced logic tricks.

Merged changes by kbranch to allow booleans in logic check and upstream changes including adding rooster jesus jump as a variable.

Last commit is a merge issue where I accidently deleted text_clip variable on merge with upstream.

### CODE REVIEW

**Comment by foxsouns** [comment]
scrolled through the changes, lots of "namesake" things, but i dont think that's bad. all checks seem to pass, so it LGTM.

\- "the fox who found jesus rooster"
"a '1/5th chance of working' 1 frame window input sounds great"

File: logic/requirements.py [hunk 1]
```diff
@@ -266,6 +268,41 @@ def __init__(self, options):
...
+        self.hookshot_clip_block = HOOKSHOT # use hookshot spam with enemies to clip through entire blocks (d5 room before gohma, d2 pots room before boss)
+        self.hookshot_over_pit = HOOKSHOT # use hookshot while over a pit to reach certain areas (see d3 vacuum room, d5 north of crossroads for example)
+        self.hookshot_jump = AND(HOOKSHOT, FEATHER) # while over pits, on the first frame after the hookshot is retracted you can input a jump to cross big pit gaps
+        self.bookshot = AND(FEATHER, HOOKSHOT) # use feather on A, hookshot on B on the same frame to get a speedy hookshot that can be used to clip past blocks
+        self.bomb_trigger = BOMB # drop two bombs at the same time to trigger cutscenes or pickup items (can use pits, or screen transitions
+        self.shield_bump = SHIELD # use shield to knock back enemies or knock off enemies when used in combination with superjumps
+        self.text_clip = options.nagmessages # trigger a text box on keyblock or rock or obstacle while holding diagonal to clip into the side. Not sure how this works, during testing it looks to be working as intended for logic
```
Comment (daid) [inline logic/requirements.py:304]: By making nag messages part of the logic, they will effect seeds and thus are not strictly part of the "user options" anymore. As everything in user options should be changeable without changing the seed.

File: logic/requirements.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def hasConsumableRequirement(self) -> bool:
 
     def test(self, inventory) -> bool:
         for item in self.__items:
-            if item in inventory:
+            if item == True or item in inventory:
```
Comment (daid) [inline logic/requirements.py:39]: As this is part of an "inner loop" of the randomizer, this change might have a performance impact. With this many changes, I'm also not clearly seeing why it is needed.

Comment (dragonc0) [reply logic/requirements.py:39]: think this was mainly to include the use of booleans like with text_clip into the logic file. @kbranch do you know more?

Comment (kbranch) [reply logic/requirements.py:39]: Oh, huh, I probably messed up a merge somewhere, that looks like the original implementation of booleans in requirements. I'll try to sort it out and send Dragon a PR.

File: logic/requirements.py (hunk 1 above)
Comment (kbranch) [reply logic/requirements.py:302]: That's unfortunate, I hadn't thought of that either. I could see people changing the nag messages setting without thinking it'd change the seed, even if moved out of the user options group.

Between the two not-great options, I'd vote for removing text clips from logic, personally.

Comment (daid) [reply logic/requirements.py:302]: yeah, I think removing text clips from logic might be the best move.
I'm also concerned about the text clips with rocks/pots, as that message no longer shows up after you finished D2.

File: logic/requirements.py [hunk 3]
```diff
@@ -299,7 +299,7 @@ def __init__(self, options):
         self.bookshot = AND(FEATHER, HOOKSHOT) # use feather on A, hookshot on B on the same frame to get a speedy hookshot that can be used to clip past blocks
         self.bomb_trigger = BOMB # drop two bombs at the same time to trigger cutscenes or pickup items (can use pits, or screen transitions
         self.shield_bump = SHIELD # use shield to knock back enemies or knock off enemies when used in combination with superjumps
-        self.text_clip = options.nagmessages # trigger a text box on keyblock or rock or obstacle while holding diagonal to clip into the side. Not sure how this works, during testing it looks to be working as intended for logic
+        self.text_clip = False & options.nagmessages # trigger a text box on keyblock or rock or obstacle while holding diagonal to clip into the side. Removed from logic for now
```
Comment (dragonc0) [inline logic/requirements.py:302]: Removed text clips from logic, commented out the relevant logic in dungeon/overworld as well to not cause confusion

**Pull request merged**